### PR TITLE
Support signing tx on mutiple networks

### PR DESCRIPTION
**Pull request by vanphandinh**
Update the "@o1labs/client-sdk" version for the signing feature can work on the mainnet

### CODE REVIEW

**Comment by jrwashburn** [comment]
I don't want to update to this (1.0.0) until we have the new foundation rules put in place; that way there is no chance of pushing bad calculations to the production network. This should be updated over the weekend.

File: package.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
     "typescript": "^4.1.5"
   },
   "dependencies": {
-    "@o1labs/client-sdk": "^0.2.5",
+    "@o1labs/client-sdk": "^1.0.0",
```
Comment (jrwashburn) [inline package.json:25]: Need to add support for both libraries and separate devnets from mainnet. dev/test nets need to be pinned at 0.2.5, only mainnet runs on 1.0.0.

**Comment by vanphandinh** [comment]
> I don't want to update to this (1.0.0) until we have the new foundation rules put in place; that way there is no chance of pushing bad calculations to the production network. This should be updated over the weekend.

Shall we have the new foundation rules? What is it for?

**Comment by jrwashburn** [comment]
> Shall we have the new foundation rules? What is it for?
See #21 and https://minaprotocol.com/docs/advanced/foundation-delegation-program 

These new rules require a complete redesign of the payout algorithm, which is getting started after #27 is merged - hopefully later tonight or tomorrow am. If you want to take a shot at it, let's do a walk through the algo? Otherwise, we are expecting to have a new version ready for testing on Saturday pm / Sunday am.

**Comment by vanphandinh** [comment]
> > Shall we have the new foundation rules? What is it for?
> > See #21 and https://minaprotocol.com/docs/advanced/foundation-delegation-program
> 
> These new rules require a complete redesign of the payout algorithm, which is getting started after #27 is merged - hopefully later tonight or tomorrow am. If you want to take a shot at it, let's do a walk through the algo? Otherwise, we are expecting to have a new version ready for testing on Saturday pm / Sunday am.

oh, great, I saw it :D

**Comment by vanphandinh** [comment]
@jrwashburn let take a look at it, I've pushed a commit to support multiple networks

File: src/index.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { getPayouts, PayoutDetails, PayoutTransaction } from "./core/payouts";
 import { getStakes } from "./core/stakes";
 import { getBlocks, getLatestHeight } from "./core/queries";
 
-import CodaSDK, { keypair } from "@o1labs/client-sdk";
+import { keypair } from "mainnet-client-sdk";
```
Comment (jrwashburn) [inline src/index.ts:5]: Assume keypair is stable across - find to just use from mainnet sdk? Seems okay.

**Review by jrwashburn — APPROVED** [review]
Looks good, and sorry for the hold up! Now that the new algorithm is in this should go too!

**Comment by jrwashburn** [comment]
Do you mind patching for the merge conflicts and then this can go in

File: src/index.ts (hunk 2 above)
Comment (vanphandinh) [reply src/index.ts:5]: I think the `keypair`and `payment` type are not changed so I reused it from the mainnet SDK client.
Do you have any method to choose the `type` conditionally?

**Comment by vanphandinh** [comment]
I've just pushed a commit to fix conflicts, but I see on your new changes, do we have an option to just sign it without broadcast? I think we can add a variable on the `.env` file to let users can choose just signing or both signing and broadcasting

**Comment by jrwashburn** [comment]
> I've just pushed a commit to fix conflicts,
> 
> But I see on your new changes, do we have an option to just sign it without broadcast? I think we can add a variable on the `.env` file to let users can choose just signing or both signing and broadcasting

I will add an issue for that. How are you planning to broadcast them though?

**Comment by jrwashburn** [comment]
Merged too soon; doesn't handle graphql, etc. reverted.

**Comment by vanphandinh** [comment]
> > I've just pushed a commit to fix conflicts,
> > But I see on your new changes, do we have an option to just sign it without broadcast? I think we can add a variable on the `.env` file to let users can choose just signing or both signing and broadcasting
> 
> I will add an issue for that. How are you planning to broadcast them though?

It’s just an option for anyone who wants to review their distribution before broadcast it.

**Pull request merged**